components/raftstore/src/store/peer.rs

```diff
@@ -3205,6 +3211,31 @@ where
                 };
             }
         }
+        let flags = WriteBatchFlags::from_bits_check(req.get_header().get_flags());
+        if flags.contains(WriteBatchFlags::STALE_READ) {
+            let read_ts = decode_u64(&mut req.mut_header().take_flag_data().as_slice()).unwrap();
+            let safe_ts = self.safe_ts.load(Ordering::Relaxed);
```

Comment:
Why use `Relaxed`?

Reply:
`safe_ts` just an incrementing counter for now, and only needs an atomic read.

Reply:
Why? I think it stands for "some data are written before this ts". Using `Relaxed` no ordering is guaranteed.